3rdparty/mshadow/mshadow/extension/reduce_with_axis.h

```diff
@@ -112,7 +112,7 @@ struct Plan<ReduceWithAxisExp<Reducer, SrcExp, DType, dimsrc, mask, dimdst>, DTy
         index_t z = (x*size_+k)*trailing_+y;
         DType tmp = res;
         Reducer::Reduce(res, src_.Eval(z/last_, z%last_));
-        if (tmp != res) {
+        if (tmp != res && !isnan_typed::IsNan(tmp)) {
```

Comment:
`NaN` is treated as greater/less than all other values in argmax/ argmin.
NaN is not equal to any number including NaN.

Reply:
In C++ i found using gdb that Nan are treated as largest. You probably don't need to check that. But as per C++ documentation the behavior is not defined so its good to keep that check !

Reply:
It is necessary to check NaN. For example, argmin/argmax should return the first index in the array [NaN, NaN], however (NaN != NaN) is true.